core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -0,0 +1,174 @@
+<script type="text/ng-template" id="components/topNavigationBar">
```

Comment:
The entire top navigation bar would be in a single directive, just with child directives making it easier to understand and maintain. This is actually where the folder structure becomes effective. The directory name is of the top-level sidebar directive other directives will use, whereas other directives within the folder are used to contain pieces of the navbar directive.